File: elearning_web/lib/models/course.dart
class Course {
  final int? id;
  final String title;
  final String description;
  final String? category;
  final String? videoUrl;
  final String? pdfUrl;

  Course({
    this.id,
    required this.title,
    required this.description,
    this.category,
    this.videoUrl,
    this.pdfUrl,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['id'] is int ? json['id'] : (json['id'] != null ? int.parse(json['id'].toString()) : null),
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      category: json['category'],
      videoUrl: json['videoUrl'],
      pdfUrl: json['pdfUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'title': title,
      'description': description,
      if (category != null) 'category': category,
      if (videoUrl != null) 'videoUrl': videoUrl,
      if (pdfUrl != null) 'pdfUrl': pdfUrl,
    };
  }
}

File: src/main/java/com/example/elearning/entities/Course.java
package com.example.elearning.entities;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "course")
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String category;

    @Column(name = "video_url", columnDefinition = "TEXT")
    private String videoUrl;

    @Column(name = "pdf_url", columnDefinition = "TEXT")
    private String pdfUrl;

    @ManyToOne
    @JoinColumn(name = "teacher_id")
    private User teacher;
    @OneToMany(mappedBy = "course", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonManagedReference

    private List<Chapter> chapters;
//    @ManyToOne
//    @JoinColumn(name = "course_id")
//    private Course course;

}
